### PR TITLE
chore: CON-1257 Rename `ecdsa` modules, `EcdsaClient`, `EcdsaGossip` and `EcdsaImpl`

### PR DESCRIPTION
**Pull request by eichhorl**
These structs and modules should be renamed because they are now also used for threshold Schnorr.

There are no alerts for the renamed metrics:
- ecdsa_on_state_change_duration_seconds
- ecdsa_client_metrics
- ecdsa_client_errors
- ecdsa_priority_fn_dropped_adverts

Affected dashboards will be updated in a separate MR

### CODE REVIEW

**Review by basvandijk — APPROVED** [review]

**Review by rumenov — APPROVED** [review]

File: rs/consensus/src/idkg/utils.rs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 //! Common utils for the ECDSA implementation.
```
Comment (kpop-dfinity) [inline rs/consensus/src/idkg/utils.rs:1]: nit: comment

File: rs/consensus/src/idkg.rs [hunk 2]
```diff
@@ -234,9 +234,9 @@ const LOOK_AHEAD: u64 = 10;
 /// Frequency for clearing the inactive key transcripts.
 pub(crate) const INACTIVE_TRANSCRIPT_PURGE_SECS: Duration = Duration::from_secs(60);
 
-/// `EcdsaImpl` is the consensus component responsible for processing threshold
+/// `IDkgImpl` is the consensus component responsible for processing threshold
 /// ECDSA payloads.
```
Comment (kpop-dfinity) [inline rs/consensus/src/idkg.rs:238]: nit: comment

File: rs/consensus/src/idkg.rs [hunk 3]
```diff
@@ -245,13 +245,13 @@ pub struct EcdsaImpl {
...
-    metrics: EcdsaClientMetrics,
+    metrics: IDkgClientMetrics,
     logger: ReplicaLogger,
     #[cfg_attr(not(feature = "malicious_code"), allow(dead_code))]
     malicious_flags: MaliciousFlags,
 }
 
-impl EcdsaImpl {
+impl IDkgImpl {
     /// Builds a new threshold ECDSA component
```
Comment (kpop-dfinity) [inline rs/consensus/src/idkg.rs:255]: nit: comment

File: rs/consensus/src/idkg.rs [hunk 4]
```diff
@@ -40,7 +40,7 @@
 //! Complete transcripts will be included in blocks via the functions
 //! `create_data_payload` and `create_summary_payload`.
 //!
-//! # [EcdsaImpl] behavior
+//! # [IDkgImpl] behavior
 //! The ECDSA component is responsible for adding artifacts to the ECDSA
```
Comment (kpop-dfinity) [inline rs/consensus/src/idkg.rs:44]: nit: the comment needs to be adjusted

**Review by kpop-dfinity — APPROVED** [review]
There are several comments which, I think, should be adjusted.
Are you planning to `grep` through the consensus directory to see if there are more comments which mention `ECDSA` but should mention something more general?

**Comment by eichhorl** [comment]
> There are several comments which, I think, should be adjusted. Are you planning to `grep` through the consensus directory to see if there are more comments which mention `ECDSA` but should mention something more general?

Yes, I'm planning to update comments separately

**Comment by kpop-dfinity** [comment]
> > There are several comments which, I think, should be adjusted. Are you planning to `grep` through the consensus directory to see if there are more comments which mention `ECDSA` but should mention something more general?
> 
> Yes, I'm planning to update comments separately

Great, thanks. You can ignore my other comments then

**Pull request merged**